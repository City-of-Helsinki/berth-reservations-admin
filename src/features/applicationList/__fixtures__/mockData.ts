import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
