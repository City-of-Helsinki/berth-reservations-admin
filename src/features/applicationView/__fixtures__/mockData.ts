import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
