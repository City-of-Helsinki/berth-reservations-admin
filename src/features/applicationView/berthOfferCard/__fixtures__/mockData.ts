import { BerthLease_lease as BerthLease } from '../__generated__/BerthLease';
import {
  AdditionalProductType,
  BerthMooringType,
  LeaseStatus,
  ProductServiceType,
} from '../../../../@types/__generated__/globalTypes';

export const lease: BerthLease = {
  id: '165023f5-5b74-4656-9315-d4e1a22f111d',
  status: LeaseStatus.DRAFTED,
  berth: {
    depth: null,
    length: 6,
    mooringType: BerthMooringType.SINGLE_SLIP_PLACE,
    width: 2.5,
    comment: '',
    isAccessible: null,
    number: '1',
    pier: {
      properties: {
        identifier: 'C',
        electricity: true,
        gate: true,
        lighting: true,
        mooring: true,
        wasteCollection: true,
        water: true,
        harbor: {
          id: 'b55da996-cf00-4b52-9aa0-6b13fd8d806b',
          properties: {
            name: 'Aurinkolahden venesatama (Aurinkoranta)',
            __typename: 'HarborProperties',
          },
          __typename: 'HarborNode',
        },
        __typename: 'PierProperties',
      },
      __typename: 'PierNode',
    },
    __typename: 'BerthNode',
  },
  customer: {
    __typename: 'ProfileNode',
    id: 'MOCK-CUSTOMER',
    primaryEmail: {
      __typename: 'EmailNode',
      id: 'MOCK-EMAIL',
      email: 'test@example.com',
    },
  },
  order: {
    id: '07f65f54-4a25-47fd-9eea-d9fdddfa46f9',
    orderNumber: '123456',
    price: '96.67',
    totalPrice: '256.70',
    orderLines: {
      edges: [
        {
          node: {
            id: '9b97b9bc-bba5-46d9-aae2-86e8686012ef',
            price: '20.50',
            product: {
              id: 'dc7f8158-a4cb-49a9-9368-e586f224fdb5',
              service: ProductServiceType.ELECTRICITY,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ea2ef175-f790-45a0-b16f-9e5a98340691',
            price: '19.33',
            product: {
              id: '2e55f605-061e-4299-814d-bf2cd53aa8ce',
              service: ProductServiceType.WATER,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '8ba4de61-63cc-49fc-afe2-26068f3c8e4d',
            price: '2.42',
            product: {
              id: '3a2e3a2f-2782-40e9-a139-b9a4d4a36c3b',
              service: ProductServiceType.GATE,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '59c768c1-f371-4178-afda-1e16d2c725b0',
            price: '8.33',
            product: {
              id: '762da6e9-89be-43bc-b097-53dff4ed8415',
              service: ProductServiceType.MOORING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '6d910d8e-80b6-4486-aa85-4958bcd98dd5',
            price: '4.17',
            product: {
              id: '34c5b6bc-7eb8-457e-bce8-8ba3a07b9057',
              service: ProductServiceType.WASTE_COLLECTION,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ab7cf1e5-92b5-42ad-844a-e2d577d30baf',
            price: '40.28',
            product: {
              id: '650dbfcb-8f10-46b6-af64-197b77686a5b',
              service: ProductServiceType.LIGHTING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '244a7af9-6f91-4f08-9738-82ef06a2e790',
            price: '65.00',
            product: {
              id: '5194dd14-0226-4603-808c-831cfbec72de',
              service: ProductServiceType.DINGHY_PLACE,
              productType: AdditionalProductType.OPTIONAL_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
      ],
      __typename: 'OrderLineNodeConnection',
    },
    __typename: 'OrderNode',
  },
  __typename: 'BerthLeaseNode',
};
