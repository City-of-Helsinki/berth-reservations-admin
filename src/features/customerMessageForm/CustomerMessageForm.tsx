import * as React from 'react';
import { Formik } from 'formik';
import { TextArea, TextInput } from 'hds-react';
import { TFunction } from 'i18next';
import * as Yup from 'yup';
import { ObjectSchema } from 'yup';
import { useTranslation } from 'react-i18next';

import { MessageFormValues } from './types';
import styles from './customerMessageForm.module.scss';
import Select from '../../common/select/Select';
import Button from '../../common/button/Button';
import Grid from '../../common/grid/Grid';
import Preview from './preview/Preview';

export type CustomerMessageFormProps = {
  handleCancel: () => void;
  handleCancelPreview: () => void;
  handlePreview: (templateId: string) => void;
  handleSendMessage?: (message: MessageFormValues) => void;
  previewHtml: string | undefined;
  recipientCount: number;
  templateOptions: { value: string; label: string }[];
};

const getMessageFormValidationSchema = (t: TFunction): ObjectSchema => {
  return Yup.object().shape({
    subject: Yup.string().required(t('forms.common.errors.required')),
    message: Yup.string().required(t('forms.common.errors.required')),
  });
};

export const CustomerMessageForm = ({
  handleCancel,
  handleCancelPreview,
  handlePreview,
  handleSendMessage,
  previewHtml,
  recipientCount,
  templateOptions,
}: CustomerMessageFormProps) => {
  const { t } = useTranslation();

  const initialValues: MessageFormValues = {
    templateId: templateOptions[0].value,
    subject: '',
    message: '',
  };

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={(values) => handleSendMessage?.(values)}
      validateOnBlur={false}
      validateOnChange={false}
      validationSchema={getMessageFormValidationSchema(t)}
    >
      {({ values, errors, handleChange, handleSubmit }) => {
        return (
          <form onSubmit={handleSubmit} className={styles.form}>
            {!previewHtml ? (
              <>
                <Grid colsCount={2}>
                  <Select
                    id="templateId"
                    value={values.templateId}
                    options={templateOptions}
                    onChange={handleChange}
                    label={t('customerList.message.template')}
                    required
                  />
                  <div className={styles.recipientCount}>
                    {t('customerList.message.recipient', { count: recipientCount })}
                  </div>
                </Grid>
                <TextInput
                  id="subject"
                  value={values.subject}
                  onChange={handleChange}
                  label={t('customerList.message.subject')}
                  invalid={!!errors.subject}
                  helperText={errors.subject}
                  required
                />
                <TextArea
                  id="message"
                  className={styles.message}
                  value={values.message}
                  onChange={handleChange}
                  label={t('customerList.message.message')}
                  invalid={!!errors.message}
                  helperText={errors.message || t('customerList.message.messageHelperText')}
                />
              </>
            ) : (
              <>
                <p>
                  {values.subject}ã€€{t('customerList.message.recipient', { count: recipientCount })}
                </p>
                <Preview html={previewHtml as string} />
                <p>
                  {`${t('customerList.message.template')}: ${
                    (templateOptions.find(({ value }) => {
                      return value === values.templateId;
                    }) as { label: string }).label
                  }`}
                </p>
              </>
            )}

            <div className={styles.formActionButtons}>
              <Button variant="secondary" onClick={handleCancel}>
                {t('common.cancel')}
              </Button>
              <div className={styles.formActionButtonsRight}>
                {!previewHtml ? (
                  <Button variant="secondary" onClick={() => handlePreview(values.templateId)}>
                    {t('customerList.message.preview')}
                  </Button>
                ) : (
                  <Button variant="secondary" onClick={handleCancelPreview}>
                    {t('common.edit')}
                  </Button>
                )}
                <Button type="submit">{t('customerList.message.send')}</Button>
              </div>
            </div>
          </form>
        );
      }}
    </Formik>
  );
};
