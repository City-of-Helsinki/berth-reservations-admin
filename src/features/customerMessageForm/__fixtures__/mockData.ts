import { NOTIFICATION_TEMPLATES } from '../__generated__/NOTIFICATION_TEMPLATES';
import { NotificationTemplateLanguage } from '../../../@types/__generated__/globalTypes';

export const mockHtml =
  // eslint-disable-next-line max-len
  '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="fi" xml:lang="fi"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width"><title>Test HTML</title></head><body><h1>Test HTML</h1><p>Test HTML</p></body></html>';

export const mockData: NOTIFICATION_TEMPLATES = {
  notificationTemplates: {
    edges: [
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjE=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: Berth application',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: båtplatsansökning',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: venepaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'berth_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjI=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: Veneen talvisäilytyspaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: ansökan om vinteruppläggningsplats',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: boat winter storage application',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'winter_storage_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
    ],
    __typename: 'NotificationTemplateNodeConnection',
  },
};
