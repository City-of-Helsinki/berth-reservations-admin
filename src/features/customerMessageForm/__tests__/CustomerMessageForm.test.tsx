import React from 'react';
import { mount } from 'enzyme';

import { CustomerMessageForm, CustomerMessageFormProps } from '../CustomerMessageForm';
import { mockHtml } from '../__fixtures__/mockData';

const mockProps: CustomerMessageFormProps = {
  handleCancel: jest.fn(),
  handleCancelPreview: jest.fn(),
  handlePreview: jest.fn(),
  handleSendMessage: jest.fn(),
  previewHtml: undefined,
  recipientCount: 1,
  templateOptions: [
    {
      value: 'MOCK-TEMPLATE',
      label: 'TestTemplate',
    },
  ],
};

describe('CustomerMessageForm', () => {
  const getWrapper = (props?: Partial<CustomerMessageFormProps>) =>
    mount(<CustomerMessageForm {...mockProps} {...props} />);

  it('renders normally in edit mode', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally in preview mode', () => {
    const wrapper = getWrapper({
      previewHtml: mockHtml,
    });

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should call handleSendMessage when form is submitted', () => {
    const handleSendMessage = jest.fn();
    const wrapper = getWrapper({
      handleSendMessage,
    });

    const onSubmit = wrapper.find('Formik').prop('onSubmit') as Function;
    onSubmit();
    wrapper.update();

    expect(handleSendMessage).toHaveBeenCalled();
  });

  it('should call handlePreview when preview button is clicked', () => {
    const handlePreview = jest.fn();
    const wrapper = getWrapper({
      handlePreview,
    });

    wrapper.find('Button').at(1).simulate('click');

    expect(handlePreview).toHaveBeenCalledWith('MOCK-TEMPLATE');
  });
});
