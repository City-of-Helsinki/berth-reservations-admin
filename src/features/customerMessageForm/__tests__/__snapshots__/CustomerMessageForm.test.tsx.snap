// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerMessageForm renders normally in edit mode 1`] = `
<form
  class="form"
>
  <h4
    class="text brand h4"
  >
    VIESTI
  </h4>
  <div
    class="grid cols2"
  >
    <div
      class="Dropdown-module_root__2LI4- select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="templateId-toggle-button"
        id="templateId-label"
      >
        Viestipohja
        <span
          class="RequiredIndicator-module_indicator__Pt7GX"
        >
          *
        </span>
      </label>
      <div
        class="Dropdown-module_wrapper__1oIw5"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="templateId-label templateId-toggle-button"
          class="Dropdown-module_buttonDropdown__IwbnQ Dropdown-module_dropdown__22ZdF"
          id="templateId-toggle-button"
          type="button"
        >
          <span
            class="Dropdown-module_buttonLabel__2jVRU"
          >
            TestTemplate
          </span>
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Dropdown-module_angleIcon__1Y1mT"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="templateId-label"
        class="Dropdown-module_menu__3rei6"
        id="templateId-menu"
        role="listbox"
        style="max-height: calc(var(--dropdown-height) * 5);"
        tabindex="-1"
      />
    </div>
    <div
      class="recipientCount"
    >
      1 vastaanottaja
    </div>
  </div>
  <div
    class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="subject"
    >
      Viestin aihe
      <span
        class="RequiredIndicator-module_indicator__Pt7GX"
      >
        *
      </span>
    </label>
    <div
      class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
    >
      <input
        class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
        id="subject"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6 message"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="message"
    >
      Viesti
    </label>
    <div
      class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
    >
      <textarea
        class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
        id="message"
      />
    </div>
    <div
      class="TextInput-module_helperText__anJHZ text-input_hds-text-input__helper-text__X1cCW"
    >
      Voit tarkastella viestiä esikatselussa ennen lähetystä.
    </div>
  </div>
  <div
    class="formActionButtons"
  >
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
      type="button"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Peruuta
      </span>
    </button>
    <div
      class="formActionButtonsRight"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Esikatsele
        </span>
      </button>
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="submit"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Lähetä
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`CustomerMessageForm renders normally in preview mode 1`] = `
<form
  class="form"
>
  <h4
    class="text brand h4"
  >
    VIESTI
  </h4>
  <p>
    　1 vastaanottaja
  </p>
  <div
    class="preview"
  >
    <iframe
      class="iframe"
      src="data:text/html,%3C!DOCTYPE%20html%20PUBLIC%20%22-%2F%2FW3C%2F%2FDTD%20XHTML%201.0%20Transitional%2F%2FEN%22%20%22http%3A%2F%2Fwww.w3.org%2FTR%2Fxhtml1%2FDTD%2Fxhtml1-transitional.dtd%22%3E%3Chtml%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxhtml%22%20lang%3D%22fi%22%20xml%3Alang%3D%22fi%22%3E%3Chead%3E%3Cmeta%20http-equiv%3D%22Content-Type%22%20content%3D%22text%2Fhtml%3B%20charset%3Dutf-8%22%3E%3Cmeta%20name%3D%22viewport%22%20content%3D%22width%3Ddevice-width%22%3E%3Ctitle%3ETest%20HTML%3C%2Ftitle%3E%3C%2Fhead%3E%3Cbody%3E%3Ch1%3ETest%20HTML%3C%2Fh1%3E%3Cp%3ETest%20HTML%3C%2Fp%3E%3C%2Fbody%3E%3C%2Fhtml%3E"
      title="PreviewModal"
    />
  </div>
  <p>
    Viestipohja: TestTemplate
  </p>
  <div
    class="formActionButtons"
  >
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
      type="button"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Peruuta
      </span>
    </button>
    <div
      class="formActionButtonsRight"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Muokkaa
        </span>
      </button>
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="submit"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Lähetä
        </span>
      </button>
    </div>
  </div>
</form>
`;
